src/main/java/com/cgdpd/library/dto/pagination/PagedResponse.java

```diff
@@ -0,0 +1,27 @@
+package com.cgdpd.library.dto.pagination;
+
+import static com.cgdpd.library.validation.Validator.required;
+import static com.cgdpd.library.validation.Validator.requiredNotNegative;
+import static com.cgdpd.library.validation.Validator.requiredPositive;
+
+import lombok.Builder;
+
+import java.util.List;
+
+@Builder
+public record PagedResponse<T>(List<T> content,
+                               int pageNumber,
+                               int pageSize,
+                               long totalElements) {
+
+    public PagedResponse {
+        required("content", content);
+        requiredPositive("pageNumber", pageNumber);
+        requiredNotNegative("pageSize", pageSize);
+        requiredNotNegative("totalElements", totalElements);
+    }
+
+    public int totalPages() {
```

Comment:
I just checked the `Paged` object and it contains this method `int getTotalPages();`, so why are we calculating it manually instead of receiving in the constructor?